test/unit/alphabet/semi_alphabet_test_template.hpp

```diff
@@ -0,0 +1,184 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/alphabet/exception.hpp>
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/test/pretty_printing.hpp>
+
+using namespace seqan3;
+
+template <typename T>
+using semi_alphabet_test = ::testing::Test;
+
+constexpr size_t maximum_iterations = 65536u;
+
+TYPED_TEST_CASE_P(semi_alphabet_test);
+
+TYPED_TEST_P(semi_alphabet_test, concept_check)
+{
+    EXPECT_TRUE(semialphabet<TypeParam>);
+    EXPECT_TRUE(semialphabet<TypeParam &>);
+    EXPECT_TRUE(semialphabet<TypeParam const>);
+    EXPECT_TRUE(semialphabet<TypeParam const &>);
+
+    EXPECT_TRUE(writable_semialphabet<TypeParam>);
+    EXPECT_TRUE(writable_semialphabet<TypeParam &>);
+    EXPECT_FALSE(writable_semialphabet<TypeParam const>);
+    EXPECT_FALSE(writable_semialphabet<TypeParam const &>);
+}
+
+TYPED_TEST_P(semi_alphabet_test, type_properties)
+{
+    // It is highly recommended that non-reference types that model this concept, also model:
+    EXPECT_TRUE((std::regular<TypeParam>));
+    EXPECT_TRUE((seqan3::trivially_copyable<TypeParam>));
+    EXPECT_TRUE((seqan3::standard_layout<TypeParam>));
+}
+
+TYPED_TEST_P(semi_alphabet_test, alphabet_size_)
+{
+    EXPECT_GT(alphabet_size<TypeParam>, 0u);
+}
+
+TYPED_TEST_P(semi_alphabet_test, default_value_constructor)
+{
+    [[maybe_unused]] TypeParam t1;
+    [[maybe_unused]] TypeParam t2{};
```

Comment:
fun test. Maybe check the default value of it. also if this means that you might test some other behaviour as well. But since both tests would fail it is still easy to track what might fail and what not.

Reply:
If it is not even used the compiler will very likely remove it anyway. So if it is just for testing that there is a default constructor you can do `std::is_default_constructible_v`. At least you can test for it. And instead of a compilation error you will get a failing test.